util/statvar_dcid_generator.py

```diff
@@ -0,0 +1,222 @@
+    except KeyError:
+        return None
+
+    quantity = capitalize_process_word(quantity)
+    return f'{value}{quantity}'
+
+
+def get_stat_var_dcid(stat_var_dict, ignore_props=None):
+    """Generates the dcid given a statistical variable.
+
+    Args:
+        stat_var_dict: A dictionary with property: values of the statistical
+          variable as key-value pairs.
+        ignore_props: A list of properties to ignore from stat_var_dict when
+          generating the dcid. This list of ignore_props will be added to the
+          default set of properties that are ignored. The ignore_props can be
+          used to account for dependent PVs to ignore when generating the dcid.
+          For example in the following statVar,
+          {
+            populationType: Person
+            measuredProperty: count
+            statType: measuredValue
+            healthInsurance: NoHealthInsurance
+            armedForceStatus: Civilian
+            institutionalization: USC_NonInstitutionalized
+          }
+          since the healthInsurance property indicates they are Civilian and
+          USC_NonInstitutionalized, ignore_props can be the list
+          ['armedForceStatus', 'institutionalization']. During the dcid
+          generation process, these PVs will not be considered.
+
+    Returns:
+        A string represting the dcid of the statistical variable.
+    """
+
+    # TODO: Add support for naming boolean constraints
+    # TODO: Stripping measurement properties (USC_) from constraints
+    # TODO: Replacing NAICS industry codes with the NAICS names
+    # TODO: Renaming cause of death properties
+    # TODO: Renaming DEA drug names
+    # TODO: Prepend or append text to some constraints to improve readability
+    # TODO: InsuredUmemploymentRate should become Rate_Insured_Unemployment
+
+    # Helper function to add a property to the dcid list.
+    def add_prop_to_list(prop: str, svd: dict, dcid_list: list):
+        if prop in svd:
+            token = capitalize_process_word(svd[prop])
+            if token is not None:
+                dcid_list.append(token)
+            del svd[prop]
+
+    dcid_list = list()
+    denominator_suffix = ''
+    svd = copy.deepcopy(stat_var_dict)
+
+    if ignore_props is None:
+        ig_p = _default_ignore_props
+    else:
+        ig_p = copy.deepcopy(ignore_props)
+        ig_p.extend(_default_ignore_props)
+
+    for prop in ig_p:
+        if prop in svd:
+            del svd[prop]
+
+    if 'measurementDenominator' in svd:
+        denominator_suffix = 'AsAFractionOf_' + svd['measurementDenominator']
+        del svd['measurementDenominator']
+
+    add_prop_to_list('measurementQualifier', svd, dcid_list)
+
+    if ('statType' in svd) and (svd['statType'].find("measuredValue") == -1):
+        token = capitalize_process_word(svd['statType'])
+        if token in _stat_type_lookup:
+            dcid_list.append(_stat_type_lookup[token])
+            del svd['statType']
```

Comment:
Since measuredValue can't show up here, could you just strip "Value" suffix?

That will then handle other statTypes like MaxValue, MinValue, SumValue

Reply:
Done.